FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_mutual_auth/DemoTasks/MutualAuthMQTTExample.c

```diff
@@ -0,0 +1,701 @@
+    xNetworkStatus = TLS_FreeRTOS_Connect( pxNetworkContext,
+                                           democonfigAWS_IOT_ENDPOINT,
+                                           democonfigMQTT_BROKER_PORT,
+                                           pxNetworkCredentials,
+                                           TRANSPORT_SEND_RECV_TIMEOUT_MS,
+                                           TRANSPORT_SEND_RECV_TIMEOUT_MS );
+    configASSERT( xNetworkStatus == TLS_TRANSPORT_SUCCESS );
+    LogInfo( ( "A mutually authenticated TLS connection established with %s.\r\n",
+               democonfigAWS_IOT_ENDPOINT ) );
+}
+/*-----------------------------------------------------------*/
+static void prvCreateMQTTConnectionWithBroker( MQTTContext_t * pxMQTTContext,
+                                               NetworkContext_t * pxNetworkContext )
+{
+    MQTTStatus_t xResult;
+    MQTTConnectInfo_t xConnectInfo;
+    bool xSessionPresent;
+    TransportInterface_t xTransport;
+    MQTTGetCurrentTimeFunc_t xGetTimeFunction;
+    MQTTEventCallback_t xUserCallback;
+
+    /***
+     * For readability, error handling in this function is restricted to the use of
+     * asserts().
+     ***/
+
+    /* Fill in Transport Interface send and receive function pointers. */
+    xTransport.pNetworkContext = pxNetworkContext;
+    xTransport.send = TLS_FreeRTOS_send;
+    xTransport.recv = TLS_FreeRTOS_recv;
+
+    /* Application callback for receiving incoming published and incoming acks
+     * from MQTT library. */
+    xUserCallback = prvEventCallback;
+
+    /* Time utility function to be used by MQTT library. */
+    xGetTimeFunction = prvGetTimeMs;
+
+    /* Initialize MQTT library. */
+    xResult = MQTT_Init( pxMQTTContext, &xTransport, xGetTimeFunction, xUserCallback, &xBuffer );
+    configASSERT( xResult == MQTTSuccess );
+
+    /* Many fields not used in this demo so start with everything at 0. */
+    memset( ( void * ) &xConnectInfo, 0x00, sizeof( xConnectInfo ) );
+
+    /* Start with a clean session i.e. direct the MQTT broker to discard any
+     * previous session data. Also, establishing a connection with clean session
+     * will ensure that the broker does not store any data when this client
+     * gets disconnected. */
+    xConnectInfo.cleanSession = true;
+
+    /* The client identifier is used to uniquely identify this MQTT client to
+     * the MQTT broker. In a production device the identifier can be something
+     * unique, such as a device serial number. */
+    xConnectInfo.pClientIdentifier = democonfigCLIENT_IDENTIFIER;
+    xConnectInfo.clientIdentifierLength = ( uint16_t ) strlen( democonfigCLIENT_IDENTIFIER );
+
+    /* Set MQTT keep-alive period. It is the responsibility of the application to ensure
+     * that the interval between Control Packets being sent does not exceed the Keep Alive value.
+     * In the absence of sending any other Control Packets, the Client MUST send a PINGREQ Packet. */
```

Comment:
The comment about it being the "responsibility of the application to ensure that the interval between Control packets being sent does not exceed Keep Alive" can be misleading about an interval larger than the keep-alive being an incorrect use by the application. 
The statement can be updated to simply say:
```suggestion
    /* Set MQTT keep-alive period. If the application does not send packets at an interval less than
     * the keep-alive period, the MQTT library will send PINGREQ packets. */ 
```